crates/vm/levm/src/errors.rs

```diff
@@ -211,8 +211,18 @@ pub struct TransactionReport {
 
 impl TransactionReport {
     /// Function to add gas to report without exceeding the maximum gas limit
-    pub fn add_gas_with_max(&mut self, gas: u64, max: u64) {
-        self.gas_used = self.gas_used.saturating_add(gas).min(max);
+    pub fn add_gas_with_max(&mut self, gas: u64, max: u64) -> Result<(), VMError> {
+        let new_gas_used = self
+            .gas_used
+            .checked_add(gas)
+            .ok_or(VMError::OutOfGas(OutOfGasError::MaxGasLimitExceeded))?;
```

Comment:
Nit. Take the advantages that `thiserror` brings to us.
```suggestion
            .ok_or(OutOfGasError::MaxGasLimitExceeded)?;
```

Reply:
https://github.com/lambdaclass/ethrex/pull/1345/commits/92e4848a29c731291131fc3b9b3fb353bd38b369